emf/src/test/java/com/powsybl/emf/IGMmergeTests.java

```diff
@@ -0,0 +1,237 @@
+    }
+
+    private static void exportNetwork(Network network, Path outputDir, String baseName, Map<String, Network> validNetworks, Set<String> profilesToExport) {
+        Objects.requireNonNull(network);
+        Path filenameEq = outputDir.resolve(baseName + "_EQ.xml");
+        Path filenameTp = outputDir.resolve(baseName + "_TP.xml");
+        Path filenameSsh = outputDir.resolve(baseName + "_SSH.xml");
+        Path filenameSv = outputDir.resolve(baseName + "_SV.xml");
+        CgmesExportContext context = new CgmesExportContext();
+        context.setScenarioTime(network.getCaseDate());
+        validNetworks.forEach((name, n) -> {
+            context.addIidmMappings(n);
+        });
+
+        if (profilesToExport.contains("EQ")) {
+            exportEquipment(network, context, filenameEq);
+        }
+        if (profilesToExport.contains("TP")) {
+            exportTopology(network, context, filenameTp);
+        }
+        if (profilesToExport.contains("SSH")) {
+            exportSteadyStateHypothesis(network, context, filenameSsh);
+        }
+        if (profilesToExport.contains("SV")) {
+            exportStateVariable(network, context, filenameSv);
+        }
+    }
+
+    private static void exportEquipment(Network network, CgmesExportContext context, Path file) {
+        try (OutputStream out = Files.newOutputStream(file)) {
+            XMLStreamWriter writer = XmlUtil.initializeWriter(true, "    ", out);
+            EquipmentExport.write(network, writer, context);
+        } catch (IOException | XMLStreamException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void exportTopology(Network network, CgmesExportContext context, Path file) {
+        try (OutputStream out = Files.newOutputStream(file)) {
+            XMLStreamWriter writer = XmlUtil.initializeWriter(true, "    ", out);
+            TopologyExport.write(network, writer, context);
+        } catch (IOException | XMLStreamException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void exportSteadyStateHypothesis(Network network, CgmesExportContext context, Path file) {
+        try (OutputStream out = Files.newOutputStream(file)) {
+            XMLStreamWriter writer = XmlUtil.initializeWriter(true, "    ", out);
+            SteadyStateHypothesisExport.write(network, writer, context);
+        } catch (XMLStreamException | IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void exportStateVariable(Network network, CgmesExportContext context, Path file) {
+        try (OutputStream out = Files.newOutputStream(file)) {
+            XMLStreamWriter writer = XmlUtil.initializeWriter(true, "    ", out);
+            StateVariablesExport.write(network, writer, context);
+        } catch (IOException | XMLStreamException e) {
+            throw new RuntimeException(e);
+        }
+    }
```

Comment:
Those methods can be refactor by adding a `Runnable` in parameters

Reply:
I did it with a consumer functionnal interface.